src/strategies/flex_window.py

```diff
@@ -84,170 +86,321 @@ def step(self, event_list=[]):
+                ) - sum(cur_feed_in.values())
             else:
-                ext_load = gc.get_avg_ext_load(cur_time, self.interval) - sum(cur_feed_in.values())
-            timesteps.append({
-                "power": cur_max_power - ext_load,
-                "ext_load": ext_load,
-                "window": cur_window,
-            })
-
-        if util.get_cost(1, gc.cost) <= self.PRICE_THRESHOLD:
-            # charge max
-            commands = {}
-            for vehicle in self.world_state.vehicles.values():
-                cs_id = vehicle.connected_charging_station
-                if cs_id is None:
-                    continue
-                cs = self.world_state.charging_stations[cs_id]
-                p = gc.cur_max_power - gc.get_current_load()
-                p = util.clamp_power(p, vehicle, cs)
-                avg_power = vehicle.battery.load(self.interval, p)['avg_power']
-                commands[cs_id] = gc.add_load(cs_id, avg_power)
-                cs.current_power += avg_power
-        elif self.LOAD_STRAT == 'balanced':
-            commands = self.distribute_balanced(timesteps)
+                ext_load = gc.get_avg_ext_load(cur_time, self.interval) - sum(
+                    cur_feed_in.values()
+                )
+            # save infos for each timestep
+            vehicles = sorted(
+                [
+                    v
+                    for v in self.world_state.vehicles.values()
+                    if v.connected_charging_station is not None
+                ],
+                key=self.sort_key,
+            )
+
+            timesteps.append(
+                {
+                    "timestep_idx": timestep_idx,
+                    "power": cur_max_power - ext_load,
+                    "ext_load": ext_load,
+                    "window": cur_window,
+                    "v_load": 0,
+                    "total_load": ext_load
+                }
+            )
+            for i, row in enumerate(timesteps):
+
+                for v in vehicles:
+                    timesteps[i].update({v.connected_charging_station: 0})
+        gc.window = timesteps[0]["window"]
+        # todo: leave price out for now
+        # if costs < price threshold: load all directly
+        # if util.get_cost(1, gc.cost) <= self.PRICE_THRESHOLD:
+        #     # charge max
+        #     commands = {}
+        #     for vehicle in self.world_state.vehicles.values():
+        #         cs_id = vehicle.connected_charging_station
+        #         if cs_id is None:
+        #             continue
+        #         cs = self.world_state.charging_stations[cs_id]
+        #         p = gc.cur_max_power - gc.get_current_load()
+        #         p = util.clamp_power(p, vehicle, cs)
+        #         avg_power = vehicle.battery.load(self.interval, p)['avg_power']
+        #         commands[cs_id] = gc.add_load(cs_id, avg_power)
+        #         cs.current_power += avg_power
```

Comment:
still too much commented out code. What is planned about pricing?

Reply:
I think it should be left out for now. But maybe one wants to add it back later on, that's why I just uncommented it for now.

Reply:
I think it should be left out for now. I just thought someone might want to add it again later on, that's why I left it commented.